tool/typed_mockito_test.dart

```diff
@@ -0,0 +1,497 @@
+/// Tests for Dart Dev Compiler
+///
+/// Much of the Mockito API does not work with the Dart Dev Compiler (DDC), as
+/// it enforces type-checking in more places than dart2js. For example:
+///
+/// ```dart
+/// class A {
+///   bool fn(String) => true;
+/// }
+///
+/// when(obj.fn(any)).thenReturn(true);
+/// ```
+///
+/// In this code, `any` is certainly not a String, so it's not Strong-mode
+/// compatible, and DDC cannot work with it.
+///
+/// This file is a copy of `mockito_test.dart`, with all of the DDC-incompatible
```

Comment:
The split would be kind of arbitrary. This subset is not just the typed API; it's everything that doesn't include `any`, `captureAny`, `argThat`, and `captureArg` outside of `typed`.

I want *all* the tests to pass using `pub run test`, and this subset to pass with DDC. There's also this weird case where setters are named differently in dart Invocations vs DDC Invocations (`==` vs `=`). So the mocked setter test is different:

```dart
// test/mockito_test.dart
mock.setter = "A";
expectFail("No matching calls. All calls: MockedClass.setter=A", () {
  verify(mock.setter = "B");
});
verify(mock.setter = "A");

// tool/typed_mockito_test.dart
mock.setter = "A";
expectFail("No matching calls. All calls: MockedClass.setter==A", () {
  verify(mock.setter = "B");
});
verify(mock.setter = "A");
```